File: app/src/main/java/com/akhbulatov/githubsample/ui/main/favoritesroot/favorites/FavoritesView.java
package com.akhbulatov.githubsample.ui.main.favoritesroot.favorites;

import android.support.annotation.NonNull;

import com.akhbulatov.githubsample.models.UserDetails;
import com.akhbulatov.githubsample.ui.global.BaseView;

import java.util.List;

public interface FavoritesView extends BaseView {
    void showFavorites(@NonNull List<UserDetails> favorites);

    void showLoadingProgress(boolean show);

    void showEmptyFavorites();

    void backToUsers();
}

File: app/src/main/java/com/akhbulatov/githubsample/ui/main/favoritesroot/favorites/FavoritesView.kt
package com.akhbulatov.githubsample.ui.main.favoritesroot.favorites

import com.akhbulatov.githubsample.models.UserDetails
import com.akhbulatov.githubsample.ui.global.BaseView

interface FavoritesView : BaseView {
    fun showFavorites(favorites: List<UserDetails>)
    fun showLoadingProgress(show: Boolean)
    fun showEmptyFavorites()
    fun backToUsers()
}